### PR TITLE
prep for analyzer 5.0

### PR DESCRIPTION
**Pull request by pq**
Anticipating deprecation in analyzer 5.0.

/cc @bwilkerson @scheglov

### CODE REVIEW

File: lib/src/rules/overridden_fields.dart [hunk 1]
```diff
@@ -162,7 +162,7 @@ class _Visitor extends SimpleAstVisitor<void> {
 
     Iterable<InterfaceType> interfaces;
     if (classElement is MixinElement) {
-      interfaces = _findAllSupertypesInMixin(classElement);
+      interfaces = _findAllSupertypesInMixin(classElement as MixinElement);
```
Comment (bwilkerson) [inline lib/src/rules/overridden_fields.dart:165]: I believe that line 158 is wrong and needs to test for `InterfaceElement` (or maybe this line needs to test `enclosingElement is MixinElement` and skip the local variable `classElement`). As it is, I think that `classElement is MixinElement` will always be false.

Also, if the else block is dependent on `classElement` being a `ClassElement` then you'll need to add a test on line 166.

File: lib/src/rules/overridden_fields.dart [hunk 2]
```diff
@@ -87,7 +87,7 @@ Iterable<InterfaceType> _findAllSupertypesAndMixins(
   return interfaces.where((i) => i != interface);
 }
 
-Iterable<InterfaceType> _findAllSupertypesInMixin(ClassElement classElement) {
+Iterable<InterfaceType> _findAllSupertypesInMixin(MixinElement classElement) {
```
Comment (bwilkerson) [inline lib/src/rules/overridden_fields.dart:90]: Consider renaming the parameter to match the new type.

Comment (pq) [reply lib/src/rules/overridden_fields.dart:90]: Done.  Thanks!

File: lib/src/rules/overridden_fields.dart (hunk 1 above)
Comment (pq) [reply lib/src/rules/overridden_fields.dart]: I think it should be `InterfaceElement`.

Updated.

**Comment by pq** [comment]
Thanks for the feedback.  Should be good now.

**Review by bwilkerson — APPROVED** [review]
LGTM, but you might want to wait for Konstantin's comments before merging because he's done a lot more 4.x -> 5.0 rewrites than I have and might see something I've missed.

**Review by scheglov — APPROVED** [review]

**Pull request merged**